docs/guide/module-configuration-patterns.md

```diff
@@ -0,0 +1,332 @@
+
+## The Three Types of Module Customization
+
+ApostropheCMS provides three distinct ways to customize module behavior, each with its own purpose:
+
+### 1. Module Configuration
+
+**Use for:** Adjusting existing behavior using options that have been specifically built into modules.
+
+Configuration provides a way to set values for options that are already designed into the modules. You can only configure options that the module explicitly supports in this way, not create new ones. You can apply configuration in these places:
+
+1. In your `app.js` file when registering modules:
+
+
+<AposCodeBlock>
+
+```javascript
+export default {
+  modules: {
+    '@apostrophecms/image': {
+      options: {
+        // Simple configuration settings
+        sizes: '(min-width: 600px) 45vw, (min-width: 1140px) 530px'
+      }
+    }
+  }
+}
+```
+  <template v-slot:caption>
+    app.js
+  </template>
+</AposCodeBlock>
+
+2. In the `options` object when creating or customizing a module at the project level:
+
+<AposCodeBlock>
+
+```javascript
+export default {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    // Configuration options
+    label: 'Article',
+    pluralLabel: 'Articles',
+    sort: { publishedAt: -1 }
+  }
+}
+```
+  <template v-slot:caption>
+    modules/article/index.js
+  </template>
+</AposCodeBlock>
+
+3. In area field definitions when configuring widgets - this will set the options for this area field only:
+
+<AposCodeBlock>
+
+```javascript
+export default {
+  // ...other properties...
+  fields: {
+    add: {
+      mainContent: {
+        type: 'area',
+        options: {
+          widgets: {
+            '@apostrophecms/rich-text': {
+              // Configuration for this specific rich text widget in this area
+              toolbar: [ 'bold', 'italic', 'link' ],
+              styles: [
+                { tag: 'p', label: 'Paragraph' },
+                { tag: 'h2', label: 'Heading 2' }
+              ]
+            },
+            '@apostrophecms/image': {
+              // Configuration for this specific image widget in this area
+              size: 'full'
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+  <template v-slot:caption>
+    modules/article/index.js
+  </template>
+</AposCodeBlock>
+
+Remember: A module can only recognize configuration options that have been specifically programmed into it. You cannot invent new options and expect the module to understand them.
+
+### 2. Module Improvement
+
+**Use for:** Adding or enhancing functionality of an existing module without changing its core purpose.
+
+Module improvement lets you add new methods, event handlers, or other features to an existing module that was installed as a npm package. You do this by creating a file with the same path as the module you want to improve. For core Apostrophe modules you place your improvement files inside an `@apostrophecms` folder like they are in the `node_modules` folder:
```

Comment:
This is the point that needs to get moved up because this is the most maintainable pattern for options too.

Reply:
I didn't want to completely rearrange the flow of the document. Instead, I expanded the section on setting module configuration options to explicitly talk about path matching, without using the term 'improve'.